src/mtcp/mtcp_restart.c

```diff
@@ -496,13 +502,38 @@ mysetauxval(char **evp, unsigned long int type, unsigned long int val)
   return -1;
 }
 
-#define PAGESIZE 4096 /* We hardwire this, since we can't make libc calls. */
-// We choose this address based on our observation on CORI that it won't overlap
-// with any other memory segments. We might change this address in the future,
-// for now, it doesn't matters.
-#define vvarStartTmp ((0x40000) - 5*PAGESIZE)
-#define vdsoStartTmp ((0x40000) - 2*PAGESIZE)
+/* We hardwire these, since we can't make libc calls. */
+#define PAGESIZE 4096
+#define ROUNDADDRUP(addr, size) ((addr + size - 1) & ~(size - 1))
+// I think we're forced to use global variables, since
+// unmap_memory_areas_and_restore_vdso uses these variables subsequently.
+uint64_t vvarStartTmp = -1;
+uint64_t vdsoStartTmp = -1;
+static void
+setupTempRegions() {
+  Area area;
+  uint64_t prev_addr = 0x10000;
+  int mapsfd = mtcp_sys_open2("/proc/self/maps", O_RDONLY);
+  if (mapsfd < 0) {
+    MTCP_PRINTF("error opening /proc/self/maps; errno: %d\n", mtcp_sys_errno);
+    mtcp_abort();
+  }
+
+  while (mtcp_readmapsline(mapsfd, &area)) {
+    if (prev_addr + 5 * PAGESIZE <= area.addr) {
+      vvarStartTmp = prev_addr;
+      vdsoStartTmp = prev_addr + 3 * PAGESIZE;
```

Comment:
Why choose 5 and 3 here? Please add comments to explain that

And if they're always 2 pages apart, you only need 1 variable I think.

Reply:
using two might make it clearer for the reader, especially later on when we are comparing addresses. agreed on the additional comments, though.